### PR TITLE
Optimize some `zio.http.endpoint.Endpoint` code

### PR DESCRIPTION
**Pull request by guizmaii**
Replace the `NonEmptyChunk.formChunk(...).getOrElse(...)` with `chunk.nonEmptyOrElse(...)(...)`
Removes an `Option` allocation

### CODE REVIEW

**Comment by 987Nabil** [comment]
Please reopen when it builds

**Comment by guizmaii** [comment]
@987Nabil Should be good now :)

File: zio-http/shared/src/main/scala/zio/http/endpoint/Endpoint.scala [hunk 1]
```diff
@@ -55,7 +55,8 @@ final case class Endpoint[PathInput, Input, Err, Output, Auth <: AuthType](
   codecError: HttpCodec[HttpCodecType.ResponseType, HttpCodecError],
   documentation: Doc,
   authType: Auth,
-) { self =>
+) {
```
Comment (987Nabil) [inline zio-http/shared/src/main/scala/zio/http/endpoint/Endpoint.scala]: Unrelated and unnecessary format change

File: zio-http/shared/src/main/scala/zio/http/endpoint/Endpoint.scala [hunk 2]
```diff
@@ -257,34 +259,35 @@ final case class Endpoint[PathInput, Input, Err, Output, Auth <: AuthType](
...
-        } { case () =>
-          Left("Unsupported")
-        }
-      case AuthType.Custom(codec)       =>
-        codec.transformOrFailRight[Unit](_ => ())(_ => Left("Unsupported"))
-      case AuthType.Or(auth1, auth2, _) =>
-        authCodec(auth1).orElseEither(authCodec(auth2))(Alternator.leftRightEqual[Unit])
-    }
+    def authCodec(authType: AuthType): HttpCodec[HttpCodecType.RequestType, Unit] =
+      authType match {
```
Comment (987Nabil) [inline zio-http/shared/src/main/scala/zio/http/endpoint/Endpoint.scala]: Unrelated and unnecessary format change

**Review by 987Nabil — CHANGES_REQUESTED** [review]

File: zio-http/shared/src/main/scala/zio/http/endpoint/Endpoint.scala (hunk 1 above)
Comment (guizmaii) [reply zio-http/shared/src/main/scala/zio/http/endpoint/Endpoint.scala]: Fixed

File: zio-http/shared/src/main/scala/zio/http/endpoint/Endpoint.scala (hunk 2 above)
Comment (guizmaii) [reply zio-http/shared/src/main/scala/zio/http/endpoint/Endpoint.scala]: Fixed

File: zio-http/shared/src/main/scala/zio/http/endpoint/Endpoint.scala [hunk 3]
```diff
@@ -312,15 +311,17 @@ final case class Endpoint[PathInput, Input, Err, Output, Auth <: AuthType](
...
-        .fromChunk(handlers)
-        .getOrElse(
-          NonEmptyChunk(
-            Handler.fail(zio.http.Response(status = Status.NotFound)) -> HttpCodec.Fallback.Condition.IsHttpCodecError,
-          ),
+    def handlers2(
+      handlers: Chunk[(Handler[Env, Nothing, Request, Response], HttpCodec.Fallback.Condition)],
+    ): NonEmptyChunk[(Handler[Env, Response, Request, Response], HttpCodec.Fallback.Condition)] = {
+      def noFound: NonEmptyChunk[(Handler[Env, Response, Request, Response], HttpCodec.Fallback.Condition)] =
+        NonEmptyChunk(
```
Comment (987Nabil) [inline zio-http/shared/src/main/scala/zio/http/endpoint/Endpoint.scala:318]: Can't we make this a constant?

**Review by 987Nabil — CHANGES_REQUESTED** [review]

**Review by 987Nabil — APPROVED** [review]

**Pull request merged**